src/cmdline.rs

```diff
@@ -146,6 +154,62 @@ pub fn parse_args() -> Result<Config> {
                         .takes_value(true),
                 ),
         )
+        .subcommand(
+            SubCommand::with_name("download")
+                .about("Download a CoreOS image")
```

Comment:
I think this is the same situation as with `install`: the core of the `download` command is generic, but the data sources are not.  I was assuming we'd add additional command-line options to support RHCOS data sources.  We could have `download-fcos` and `download-rhcos`, but that seems kind of unpleasant, especially if there's eventually another downstream distro that uses FCOS- or RHCOS-compatible data sources.

Reply:
Yes (and `list-stream` too). I'm just fearing that, by the time we get there, FCOS and RHCOS metadata will have diverged enough to have a different matrix of options.
If we are confident we aren't going in that direction (I'm not on top of RHCOS pipeline/metadata right now), then my concern is moot.

Reply:
I'm not at all confident that we're not going in that direction.  I've been assuming that we'll "just" add more options for RHCOS.

I do expect `list-stream` to be FCOS-specific, FWIW.

Reply:
Ack. As our crystal balls seems to be equally out of order, I think it's a fair approach to go with what we have in this PR (also because short-term UX is better).
If later we ever found out we cannot fit RHCOS in this command anymore, we'll figure out some split or a better idea. Today is not that day.